spring-webclient-mocking/src/test/java/com/baeldung/webclientmocking/service/EmployeeServiceMockWebServerTest.java

```diff
@@ -0,0 +1,124 @@
+package com.baeldung.webclientmocking.service;
+
+import com.baeldung.webclientmocking.domain.Employee;
+import com.baeldung.webclientmocking.enums.Role;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.Rule;
+import org.junit.jupiter.api.*;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class EmployeeServiceMockWebServerTest {
+
+    public  static MockWebServer mockBackEnd=null;
+    private EmployeeService employeeService;
+    ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void setUp() throws IOException {
+        mockBackEnd = new MockWebServer();
+        mockBackEnd.start();
+    }
+
+    @AfterAll
+    static void tearDown() throws IOException {
+        mockBackEnd.shutdown();
+    }
+
+    @BeforeEach
+    void initialize(){
+
+        final String baseUrl=String.format("http://localhost:%s", mockBackEnd.getPort());
+        employeeService = new EmployeeService(baseUrl);
+        objectMapper = new ObjectMapper();
+    }
+
+    @Test
+    void getEmployeeById() throws Exception {
+
+        Integer employeeId = 100;
```

Comment:
I'm not sure why you're defining this here. If you used it in place of the other `100`, maybe it would make sense to define it up here. In my view, either use it across the whole test or don't bother with it. It might be clearer to remove it completely.